Paintroid/src/main/java/org/catrobat/paintroid/dialog/SaveBeforeFinishDialog.java

```diff
@@ -65,7 +70,14 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
 					@Override
 					public void onClick(DialogInterface dialog, int id) {
 						MainActivityContracts.Presenter presenter = activity.getPresenter();
-						presenter.saveImageConfirmClicked(requestCode, uri);
+						switch (requestCode) {
```

Comment:
This moves logic from the presenter into the `SaveBeforeFinishDialog`.
Please do this translation from `SaveImageRequestCode` to `PermissionRequestCode` in the presenter, keep the dialog as dumb as possible.